### PR TITLE
Add lint and publish check CI jobs.

### PR DESCRIPTION
**Pull request by adlerjohn**
Adds 2 jobs:
1. toml linter to make sure the toml is formatted correctly.
2. version checker on publish. When publishing a release, will make sure the `version` in the manifest file matches the release version.

### CODE REVIEW

**Review by binggh — APPROVED** [review]

File: .github/scripts/verify_tag.sh [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+fi
+
+if [ -z "$MANIFEST" ]; then
+    err "Expected manifest to be set"
+    exit 1
+fi
+
+# strip preceeding 'v' if it exists on tag
+REF=${REF/#v}
+TOML_VERSION=$(cat $MANIFEST | dasel -r toml 'package.version')
```
Comment (mitchmindtree) [inline .github/scripts/verify_tag.sh:28]: Where did you grab this script from? I thought the sway repo, but in the sway repo version this line is:
```suggestion
TOML_VERSION=$(toml get $MANIFEST package.version | tr -d '"')
```
Doesn't matter, but I wonder if one contains a fix the other is missing :thinking:

**Review by mitchmindtree — APPROVED** [review]

File: .github/scripts/verify_tag.sh (hunk 1 above)
Comment (binggh) [reply .github/scripts/verify_tag.sh:28]: fuel-core i think?

Comment (binggh) [reply .github/scripts/verify_tag.sh:28]: https://github.com/FuelLabs/fuel-core/pull/160

Comment (adlerjohn) [reply .github/scripts/verify_tag.sh:28]: `fuel-core`. The one with dasel is faster since it's a binary. Hence requesting @Voxelot review!

**Pull request merged**